pkg/models/job.go

```diff
@@ -0,0 +1,187 @@
+//go:generate stringer -type=JobStateType --trimprefix=JobStateType --output job_string.go
+package models
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/bacalhau-project/bacalhau/pkg/lib/validate"
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/exp/maps"
+)
+
+type JobStateType int
+
+const (
+	JobStateTypeUnknown JobStateType = iota
+
+	// JobStateTypePending is the state of a job that has been submitted but not
+	// yet scheduled.
+	JobStateTypePending
+
+	// JobStateTypeRunning is the state of a job that has been scheduled, with at
+	// least one active execution.
+	JobStateTypeRunning
+
+	// JobStateTypeCompleted is the state of a job that has successfully completed.
+	// Only valid for batch jobs.
+	JobStateTypeCompleted
+
+	// JobStateTypeFailed is the state of a job that has failed.
+	JobStateTypeFailed
+
+	// JobStateTypeStopped is the state of a job that has been stopped by the user.
+	JobStateTypeStopped
+)
+
+type Job struct {
+	// ID is a unique identifier assigned to this job.
+	// It helps to distinguish jobs with the same name after they have been deleted and re-created.
+	//The ID is generated by the server and should not be set directly by the client.
+	ID string
+
+	// Name is the logical name of the job used to refer to it.
+	// Submitting a job with the same name as an existing job will result in an
+	// update to the existing job.
+	Name string
+
+	// Namespace is the namespace this job is running in.
+	Namespace string
+
+	// Type is the type of job this is, e.g. "daemon" or "batch".
+	Type string
+
+	// Priority defines the scheduling priority of this job.
+	Priority int
+
+	// Count is the number of replicas that should be scheduled.
+	Count int
+
+	// Constraints is a selector which must be true for the compute node to run this job.
+	Constraints []*LabelSelectorRequirement
+
+	// Meta is used to associate arbitrary metadata with this job.
+	Meta map[string]string
+
+	Tasks []*Task
+
+	// State is the current state of the job.
+	State JobStateType
+
+	// StateDescription is meant to provide more human useful information
+	StateDescription string
+
+	// Version is a monotonically increasing version number that is incremented
+	// on each job update.
```

Comment:
```suggestion
	// Version is a per-job monotonically increasing version number that is incremented
	// on each job update.
```

After suggesting that, I'm wondering whether now the version applies to the Name/ID combo for this job, or just the ID.

Reply:
should be only for the id